feat: --onEmit (#167)
gilamran/tsc-watch: README.md

| --------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------- |
| `--onSuccess COMMAND`             | Executes `COMMAND` on **every successful** compilation.                                                                                             |
| `--onFirstSuccess COMMAND`        | Executes `COMMAND` on the **first successful** compilation.                                                                                         |

> | `--onEmit COMMAND`                | Executes debounced `COMMAND` on **every emitted file**.                                                                                             |

Reviewer: on every emitted file? are you sure?
Does this means that if I have a complication of 100 files, this command will run 100 times?

Reviewer: Also, maybe provide a use case for when someone will want to use onEmit?

Author: > Does this means that if I have a complication of 100 files, this command will run 100 times?

No because of `--onEmitDebounce`. Just added a note in the README for this.

> Also, maybe provide a use case for when someone will want to use onEmit?

Just added more to this line

> | `--onEmit COMMAND`                | Executes debounced `COMMAND` on **every emitted file**, ignoring unchanged inputs and regardless of typechecking success or failure.                |

Reviewer: ```suggestion
| `--onEmit COMMAND`                | Executes debounced `COMMAND` on **every emitted file**, ignoring unchanged files and disregards compilation success or failure.                |
```

Notes:

- The (`onSuccess`) `COMMAND` will not run if the compilation failed.
- The (`onEmit`) `COMMAND` will not run if the compilation succeeded with no changed files, unless it is the first success.

> - The (`onEmit`) `COMMAND` will run if the compilation failed, but still emitted changed files.

Reviewer: ```suggestion
- The (`onEmit`) `COMMAND` will run even if the compilation failed, but emitted changed files.
```

- The (`onEmit`) `COMMAND` will not run 100 times for 100 files, due to `--onEmitDebounce`

> - The (`onEmit`) `COMMAND` will not cancel the `onSuccess`/`onFirstSuccess`/`onFailure`/`onCompilationComplete`/`onCompilationStarted` commands and vice versa.

Reviewer: ```suggestion
- The (`onEmit`) `COMMAND` is not cancelling the `onSuccess`/`onFirstSuccess`/`onFailure`/`onCompilationComplete`/`onCompilationStarted` commands and vice versa.
```
